beacon_node/network/src/sync/range_sync/chain.rs

```diff
@@ -1092,18 +1088,17 @@ impl<T: BeaconChainTypes> SyncingChain<T> {
     ) -> bool {
         if network.chain.spec.is_peer_das_enabled_for_epoch(epoch) {
             // Require peers on all sampling column subnets before sending batches
+
             network
                 .network_globals()
                 .sampling_subnets()
                 .iter()
                 .all(|subnet_id| {
-                    let peer_db = network.network_globals().peers.read();
-                    let peer_count = self
+                    let peer_count = network
+                        .network_globals()
                         .peers
-                        .iter()
-                        .filter(|peer| {
-                            peer_db.is_good_range_sync_custody_subnet_peer(*subnet_id, peer)
-                        })
+                        .read()
+                        .good_custody_subnet_peer_range_sync(*subnet_id, epoch)
```

Comment:
How often do we call `good_peers_on_sampling_subnets`? Now it loops over all peers for all subnets (if supernode)

Reply:
Yes this is a lot - I raised this in earlier reviews, I can address this
https://github.com/sigp/lighthouse/issues/7855